### DISCUSSION TITLE
No changelog for plugin on gradle plugin portal

### DISCUSSION THREAD

**Discussion by LunNova** [renovatebot/renovate#12462] [Request Help] [answered]
I publish a gradle plugin to the gradle plugin portal: https://plugins.gradle.org/plugin/dev.minco.gradle.defaults-plugin

It's published with the website and vcsUrls set as its github repository.

It has changelogs on github: https://github.com/MinimallyCorrect/DefaultsPlugin/releases

When an update for it was detected, the generated PR did not include a changelog: https://github.com/MinimallyCorrect/Mixin/pull/60

Is this expected? Is there something I can do to make these work?

**Comment by rarkins** [comment, marked as answer]
Here's the logic for sourceUrl fetching in Maven: https://github.com/renovatebot/renovate/blob/bd71fffa30f42e27ad019441d3a2f68e49d256e1/lib/datasource/maven/util.ts#L208-L222

If you think it's complete, please raise a feature request for supporting the additional fields and include:
- link to documentation of fields
- minimal reproduction repo (creates a PR with missing changelog)

**Reply by LunNova** [reply]
It looks like the plugin marker POM https://plugins.gradle.org/m2/dev/minco/gradle/defaults-plugin/dev.minco.gradle.defaults-plugin.gradle.plugin/0.2.43/dev.minco.gradle.defaults-plugin.gradle.plugin-0.2.43.pom doesn't include the scmUrl, even though it is included in the configuration for pluginBundle.

The pom that is published to my own repo at https://maven.minco.dev/dev/minco/gradle/defaults-plugin/0.2.43/defaults-plugin-0.2.43.pom also has that set, but it seems like the gradle plugin portal is generating its own pom without this info.

Looks like an issue with the plugin portal. Found https://github.com/gradle/gradle/issues/17022

Might be able to work around it by fetching the pom for the plugin (the one dependency of the marker pom), but it doesn't seem great to add a special case specifically for the gradle plugin portal repo.

edit: Oh, it actually does work already by grabbing the pom for the plugin itself. Awesome!
